src/Sparrow/Json/AsyncBlittableJsonTextWriter.cs

```diff
@@ -61,24 +61,24 @@ public async ValueTask WriteStreamAsync(Stream stream, CancellationToken token =
         }
 
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        public ValueTask<int> MaybeFlushAsync(CancellationToken token = default)
+        public async ValueTask<int> MaybeFlushAsync(CancellationToken token = default)
         {
             var innerStream = _stream as MemoryStream;
             if (innerStream == null)
                 ThrowInvalidTypeException(_stream?.GetType());
             if (innerStream.Length * 2 <= innerStream.Capacity)
-                return new ValueTask<int>(0);
+                return 0;
 
-            FlushInternal(); // this is OK, because inner stream is a MemoryStream
-            return FlushAsync(token);
+            await FlushInternalAsync().ConfigureAwait(false); // this is OK, because inner stream is a MemoryStream
```

Comment:
does this comment is still valid?
FlushInternalAsync returns ValueTask?

Reply:
This turned out to be irrelevant to the performance. 
We can keep the `async` here or revert this change.

Reply:
Revert, lets avoid additional async machinery

Reply:
reverted